src/tests/unit/tests/common/telemetry/exception-telemetry-sanitizer.test.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import {
+    ErrorType,
+    TelemetryEventSource,
+    UnhandledErrorTelemetryData,
+} from 'common/extension-telemetry-events';
+import { ExceptionTelemetrySanitizer } from 'common/telemetry/exception-telemetry-sanitizer';
+
+describe(ExceptionTelemetrySanitizer, () => {
+    let extensionIdStub;
+    let errorMessageStub: string;
+    let errorStub: Error;
+    let telemetryStub: UnhandledErrorTelemetryData;
+    let expectedTelemetry: UnhandledErrorTelemetryData;
+    const source: TelemetryEventSource = TelemetryEventSource.AdHocTools;
+    const errorType: ErrorType = ErrorType.ConsoleError;
+
+    let testSubject: ExceptionTelemetrySanitizer;
+
+    beforeEach(async () => {
+        extensionIdStub = 'extensionId';
+        errorMessageStub = 'Error message';
+        errorStub = new Error();
+
+        testSubject = new ExceptionTelemetrySanitizer(extensionIdStub);
+    });
+
+    describe('it truncates long data', () => {
+        afterEach(() => {
+            const sanitizedTelemetry = testSubject.sanitizeTelemetryData(telemetryStub);
+
+            expect(sanitizedTelemetry).toMatchObject(expectedTelemetry);
+        });
+
+        test('it truncates messages beyond size cap', () => {
+            errorMessageStub = 'very long message'.repeat(30);
+
+            telemetryStub = {
+                message: errorMessageStub,
+                stackTrace: errorStub.stack,
+                errorType,
+                source,
+            };
+
+            expectedTelemetry = {
+                message: errorMessageStub.substring(0, 300),
+                stackTrace: errorStub.stack,
+                errorType,
+                source,
+            };
+        });
+
+        test('it truncates stack traces beyond size cap', () => {
+            errorStub.stack = 'very long stack'.repeat(500);
+
+            telemetryStub = {
+                message: errorMessageStub,
+                stackTrace: errorStub.stack,
+                errorType,
+                source,
+            };
+
+            expectedTelemetry = {
+                message: errorMessageStub,
+                stackTrace: errorStub.stack.substring(0, 5000),
+                errorType,
+                source,
+            };
+        });
+    });
+
+    describe('it returns undefined on invalid data', () => {
+        afterEach(() => {
+            const sanitizedTelemetry = testSubject.sanitizeTelemetryData(telemetryStub);
+
+            expect(sanitizedTelemetry).toBeUndefined();
+        });
+
+        test('it does not send data that includes urls', () => {
```

Comment:
It would be good to include more robust test coverage of the complicated regex matching. A technique I like to use when trying to think of test cases is to consider any bugs I find when I'm testing stuff manually (or bugs that code review finds) as good cases to add regression tests for. I often try to write the regression test before I fix the bug in those cases (that's a good verification that the test works).

For example, some test cases that would be good to add which could act as regression tests for issues we've found/discussed during the review process might be:

* It should reject cases that include `file://` URLs, extension URLs for other extensions, or non-`chrome-extension://` URLs with our extension ID as the domain
* It should accept cases where excluded properties like html appear as part of a larger string, rather than enquoted like a JSON object key
* It should reject cases that quote excluded properties in single-quotes